desktop/libs/Scripts/sync-service/index.js

```diff
@@ -0,0 +1,118 @@
+var server = require('http').createServer()
+  , url = require('url')
+  , express = require('express')
+  , app = express()
+  , cors = require('cors')
+  , WebSocketServer = require('ws').Server
+  , port = 4080;
+
+var io = require('socket.io')(4082)
+let onImage = () => {}
```

Comment:
It's a little confusing setting these to empty functions. Because they are never called. Later on there are lines like `onImage && ...`, but `onImage` always exists.

I would probably do something like

```
let onImage, onLayout, activeScene
let connected = false

io.on('connection', (socket) => {
  connected = true
  onImage = (image, sceneName) => {
    socket.emit('image', { sceneName, image })
  }
}
```

And afterward, just check `connected` before calling